pol-core/bscript/compiler/file/PrettifyFileProcessor.cpp

```diff
@@ -196,21 +197,163 @@ antlrcpp::Any PrettifyFileProcessor::visitVariableDeclarationList(
 antlrcpp::Any PrettifyFileProcessor::visitVariableDeclaration(
     EscriptParser::VariableDeclarationContext* ctx )
 {
-  make_identifier( ctx->IDENTIFIER() );
+  if ( auto identifier = ctx->IDENTIFIER() )
+  {
+    make_identifier( identifier );
+
+    if ( auto variable_declaration_initializer = ctx->variableDeclarationInitializer() )
+    {
+      _currentscope |= FmtToken::Scope::VAR;
 
-  if ( auto variable_declaration_initializer = ctx->variableDeclarationInitializer() )
+      if ( auto assign = variable_declaration_initializer->ASSIGN() )
+        addToken( ":=", assign, linebuilder.assignmentStyle() );
+
+      if ( auto expression = variable_declaration_initializer->expression() )
+        visitExpression( expression );
+      else if ( auto arr = variable_declaration_initializer->ARRAY() )
+        addToken( "array", arr, FmtToken::SPACE );
+      _currentscope &= ~FmtToken::Scope::VAR;
+    }
+  }
+  else if ( auto binding_decl = ctx->bindingDeclaration() )
   {
-    _currentscope |= FmtToken::Scope::VAR;
+    visitChildren( ctx );
+  }
 
-    if ( auto assign = variable_declaration_initializer->ASSIGN() )
-      addToken( ":=", assign, linebuilder.assignmentStyle() );
+  return {};
+}
 
-    if ( auto expression = variable_declaration_initializer->expression() )
-      visitExpression( expression );
-    else if ( auto arr = variable_declaration_initializer->ARRAY() )
-      addToken( "array", arr, FmtToken::SPACE );
-    _currentscope &= ~FmtToken::Scope::VAR;
+antlrcpp::Any PrettifyFileProcessor::visitBindingDeclaration(
+    EscriptGrammar::EscriptParser::BindingDeclarationContext* ctx )
+{
+  if ( auto lbrack = ctx->LBRACK() )
+  {
+    addToken(
```

Comment:
This should format like an array have you tested setting _currentscope |= FmtToken::Scope::ARRAY; before and remove it afterwards? (For the Indexversion it would be dict